tests/c/CMakeLists.txt

```diff
@@ -33,6 +33,11 @@ set(CMAKE_BUILD_TYPE DEBUG)
 set(CMAKE_CXX_STANDARD 17)
 SET(CMAKE_C_STANDARD 99)
 
+if((CMAKE_CXX_COMPILER_ID STREQUAL "GNU") AND (CMAKE_C_COMPILER_ID STREQUAL "GNU"))
+    set(CMAKE_CXX_FLAGS --coverage)
+    add_link_options(--coverage)
+endif()
```

Comment:
Again, I'm a bit worried about the performance impact of adding coverage globally. Can we gate it somehow to only if coverage is enabled?

Reply:
Now this is contingent on `-DCOVERAGE=on` being passed to the `cmake` command